docs/PyOp.md

```diff
@@ -0,0 +1,71 @@
+# Python Operator 
+To facilitate Python coders on model developing, onnxruntime provides a way to invoke operators implemented in Python.
+
+## Usage
+Step 1, build onnxruntime with“--config Release --enable_language_interop_ops --build_shared_lib”and override existing onnxruntime binary with the latest, then copy onnxruntime_pywrapper.dll or libonnxruntime_pywrapper.so or libonnxruntime_pywrapper.dylib to the path where onnxruntime binary is placed.
+Note that it is suggested to compile within the Python environment where inferencing will happen. For example, if inferencing will happen in a conda env named myconda1, please compile the binary within that environment as well.
+
+Step 2, create an onnx model containing Python operator nodes:
+```python
+ad1_node = helper.make_node('Add', ['A','B'], ['S'])
+mul_node = helper.make_node('Mul', ['C','D'], ['P'])
+py1_node = helper.make_node(op_type = 'PyOp', #required, must be 'PyOp'
+                            inputs = ['S','P'], #required
+                            outputs = ['L','M','N'], #required
+                            domain = 'pyopmulti_1', #required, must be unique
+                            input_types = [TensorProto.FLOAT, TensorProto.FLOAT], #required
+                            output_types = [TensorProto.FLOAT, TensorProto.FLOAT, TensorProto.FLOAT], #required
+                            module = 'mymodule', #required
+                            class_name = 'Multi_1', #required
+                            compute = 'compute', #optional, 'compute' by default
+                            W1 = '5', W2 = '7', W3 = '9') #optional, must be strings
+ad2_node = helper.make_node('Add', ['L','M'], ['H'])
+py2_node = helper.make_node('PyOp',['H','N','E'],['O','W'], domain = 'pyopmulti_2',
+                            input_types = [TensorProto.FLOAT, TensorProto.FLOAT, TensorProto.FLOAT],
+                            output_types = [TensorProto.FLOAT, TensorProto.FLOAT],
+                            module = 'mymodule', class_name = 'Multi_2')
+sub_node = helper.make_node('Sub', ['O','W'], ['F'])
+graph = helper.make_graph([ad1_node,mul_node,py1_node,ad2_node,py2_node,sub_node], 'multi_pyop_graph', [A,B,C,D,E], [F])
+model = helper.make_model(graph, producer_name = 'pyop_model')
+onnx.save(model, './model.onnx')
+```
+Step 3, implement mymodule.py:
+```python
+class Multi_1:
+    def __init__(self, W1, W2, W3):
+        self.W1 = int(W1)
+        self.W2 = int(W2)
+        self.W3 = int(W3)
+    def compute(self, S, P):
+        ret = S + P
+        return ret + self.W1, ret + self.W2, ret + self.W3
```

Comment:
What is the syntax if there is only one result?

Reply:
just return it:

return <your result>

Reply:
Coders are allowed to return arbitrary number of numpys.

Reply:
I was asking because looking at the example, I was wondering if it should be a tuple with one element or the element itself knowing that this element cannot be a tuple then.

Reply:
If only one numpy returned, Python C API will simply return an Array.